test/docfx.Test/DocfxTest.cs

```diff
@@ -19,10 +19,12 @@ namespace Microsoft.Docs.Build
     public static class DocfxTest
     {
         private static readonly JsonDiff s_jsonDiff = CreateJsonDiff();
+        private static readonly JsonDiff s_languageServerNotificationJsonDiff = CreateLanguageServerNotificationJsonDiff();
         private static readonly ConcurrentDictionary<string, object> s_locks = new ConcurrentDictionary<string, object>();
         private static readonly AsyncLocal<IReadOnlyDictionary<string, string>> t_repos = new AsyncLocal<IReadOnlyDictionary<string, string>>();
         private static readonly AsyncLocal<IReadOnlyDictionary<string, string>> t_remoteFiles = new AsyncLocal<IReadOnlyDictionary<string, string>>();
         private static readonly AsyncLocal<string> t_appDataPath = new AsyncLocal<string>();
+        private static readonly AsyncLocal<Dictionary<string, string>> t_variables = new AsyncLocal<Dictionary<string, string>>();
```

Comment:
The variables does not need to be AsyncLocal, put it to the return value of `CreateDocset` method and pass it down to the constructor of `DocfxLanguageServerTestHost` ?